crates/hulk/src/nao/hula.rs

```diff
@@ -377,10 +378,16 @@ pub struct ControlStorage {
     pub stiffness: JointsArray,
 }
 
-pub fn read_from_hula(stream: &mut UnixStream) -> Result<StateStorage> {
-    let mut read_buffer = [0; size_of::<StateStorage>()];
-    stream.read_exact(&mut read_buffer)?;
-    Ok(unsafe { read(read_buffer.as_ptr() as *const StateStorage) })
+pub fn read_from_hula<Reader, Poller>(
```

Comment:
Does this need to be generic over the `Poller`?

Reply:
Removed generic parameters of reader and poller.